tracer/src/Datadog.Trace/Activity/OtlpHelpers.cs

```diff
@@ -387,27 +391,55 @@ private static void ExtractExceptionAttributes<TInner>(TInner activity, Span spa
                     continue;
                 }
 
+                string? errorType = null;
+                string? errorMsg = null;
+                string? errorStack = null;
+
                 foreach (var tag in duckEvent.Tags)
                 {
                     switch (tag.Key)
                     {
                         case OpenTelemetryErrorType:
-                            SetTagObject(span, Tags.ErrorType, tag.Value);
+                            errorType = tag.Value?.ToString();
                             break;
 
                         case OpenTelemetryErrorMsg:
-                            SetTagObject(span, Tags.ErrorMsg, tag.Value);
+                            errorMsg = tag.Value?.ToString();
                             break;
 
                         case OpenTelemetryErrorStack:
-                            SetTagObject(span, Tags.ErrorStack, tag.Value);
+                            errorStack = tag.Value?.ToString();
                             break;
                     }
                 }
 
-                // we've found the exception attribute so we should be done here
+                // Ensure that all of the error tracking tags are updated (even null values) from the same exception attribute
+                SetTagObject(span, Tags.ErrorType, errorType);
+                SetTagObject(span, Tags.ErrorMsg, errorMsg);
+                SetTagObject(span, Tags.ErrorStack, errorStack);
+            }
+        }
+
+        internal static void SerializeEventsToJson(Span span, IEnumerable events)
+        {
+            var eventsList = new List<ActivityEvent>();
+
+            foreach (var activityEvent in events)
+            {
+                if (activityEvent.TryDuckCast<ActivityEvent>(out var duckEvent))
+                {
+                    eventsList.Add(duckEvent);
+                }
+            }
+
+            if (eventsList.Count <= 0)
+            {
                 return;
             }
+
+            var settings = new JsonSerializerSettings { Converters = new List<JsonConverter> { new ActivityEventConverter() }, Formatting = Formatting.None };
+            var eventsJson = JsonConvert.SerializeObject(eventsList, settings);
+            span.SetTag("events", eventsJson);
```

Comment:
Is there an RFC for span event support? Did we consider using an internal span tag like `_dd.events`? Or using a span field in the MessagePack payload (like span links that use `span.span_links`) instead of json in a span tag (`span.meta[events]`)?

Reply:
Yep both of those were considered in an internal RFC I can share with you 😆

In short, we want to make the information from these API's visible to the user in the short-term without any code changes in the Datadog Agent /trace retrieval / UI. So regarding `_dd.events`, we can't use that because then the tag value would not be visible to the user. And we can't use a new MessagePack payload in the short term because that would require an upgrade to the Datadog Agent

Reply:
Got it, thanks!